solr/test-framework/src/java/org/apache/solr/cloud/SolrCloudTestCase.java

```diff
@@ -72,6 +74,11 @@
  */
 public class SolrCloudTestCase extends SolrTestCaseJ4 {
 
+
+  public static <T> void assertThat(T actual, Matcher<? super T> matcher) {
```

Comment:
If having this, wondering about having it here in `SolrCloudTestCase` vs. in `SolrTestCaseJ4` vs. in `SolrTestCase` classes?

Reply:
I put it here, only to posit that this might work!  I agree that we probably want it as high up the hierachy as possible...  

DO you think this is a reasonable approach, or should we instead use explicit `MatcherAssert.assertThat` everywhere?

Reply:
Explicit `MatcherAssert.assertThat` seems quite intrusive, but that's of course subjective.

Wondering if `import static org.hamcrest.MatcherAssert.assertThat;` might provide a third alternative e.g. see https://github.com/apache/solr/blob/releases/solr/9.0.0/solr/modules/ltr/src/test/org/apache/solr/ltr/interleaving/algorithms/TeamDraftInterleavingTest.java usage?

Reply:
The static import seems better than the explicit `MatcherAssert.assertThat`.....      So...  How do we get a decision on this?  Is there anything in our code style guide that would guide us to one of these choices???

Reply:
I am tentative/hesitant:
1. I generally also prefer static imports for normal production code
2. I like Class#staticMethod more, that's also what code guidelines say: "always use class name when invoking static methods as there's no inheritance" (when you call a static method theres no inheritance involved, it is just the compiler hardcoding the class name in the call site)
3. But for testing this is quite horrible to use, because auto-completion by IDEs does not work until you have added the imports and found the correct class. By having it in the base class it is more accessible for quickly writing a test, although this is against #2.

Uwe

Reply:
So...    Am I taking it that you prefer (at least for the tests) that we override the method in the base class?   If we go all the way up the tree in Solr, it would I think be `SolrTestCase` where we would move the method?

Reply:
+1

But please remove the empty message below.

Reply:
So we can't override assertThat in `SolrTestCase` since its static and comes from `Assert` which is what `LuceneTestCase` derives from. ForbiddenApis correctly complains that we can't just override the static method and instead knows that `Assert#assertThat` is still being called. So the solution in this PR isn't going to work as is. I'm working on an updated patch to handle this.

Reply:
The static import also doesn't work in each file since the base class already has the static method `assertThat` - so the only solution is to explicitly do `MatcherAssert.assertThat`.